transform/util.go

```diff
@@ -133,6 +129,82 @@ func getJSONRaw(data []byte, path string, pathRequired bool) ([]byte, error) {
 	return result, nil
 }
 
+// setPath updates the value with properly formatted timestamp(s) and properly
+// handles array indexing
+func setJSONRaw(data, out []byte, path string) ([]byte, error) {
+	var err error
+	splitPath := strings.Split(path, ".")
+	numOfInserts := 0
+
+	for element, k := range splitPath {
+		arrayRefs := jsonPathRe.FindAllStringSubmatch(k, -1)
+		if arrayRefs != nil && len(arrayRefs) > 0 {
+			objKey := arrayRefs[0][1]      // the key
+			arrayKeyStr := arrayRefs[0][2] // the array index
+			if arrayKeyStr == "*" {
```

Comment:
Note that as implemented we do not have any transforms that ever call this code path, I don't think. I bet it was a pain in the ass to write but I'm not sure if we should support it... Basically since data is invariable it would be overwriting everything already in the array? I guess you could use it to add a new static key to every element in an array and that could be useful?

There might be some use down the road but I would consider adding a disclaimer that it isn't being used right now for maintenance purposes.